src/main/java/dev/corestone/lotrings/abilities/BonemealAbility.java

```diff
@@ -0,0 +1,50 @@
+package dev.corestone.lotrings.abilities;
+
+import dev.corestone.lotrings.LOTRings;
+import dev.corestone.lotrings.Ring;
+import dev.corestone.lotrings.RingState;
+import org.bukkit.Sound;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerInteractEvent;
+
+public class BonemealAbility extends AbilitySuper {
+
+    private Sound sound;
+
+    public BonemealAbility(LOTRings plugin, Ring ring, String abilityName) {
+        super(plugin, ring, abilityName);
+        try {
+            this.sound = Sound.valueOf(plugin.getAbilityDataManager().getAbilityStringData(abilityName, "sound").toUpperCase());
+        } catch (Exception e) {
+            sendLoadError();
+        }
+    }
+
+    @Override
+    public void switchState(RingState ringState) {
+        if (ringState == RingState.LOST) {
+            HandlerList.unregisterAll(this);
```

Comment:
If the only function of your switchState(RingState state); function is to unregister the events, you don't even need to include the function at all in your ability. The parent class, AbilitySuper.java, already unregisters events automatically. If you do add the switchState(RIngState state); function, then you will need to manually unregister it or use super to call it from the parent class. Manually unregistering like this is also fine. If this comment was confusing just ignore it.